presto-main/src/main/java/com/facebook/presto/operator/scalar/DataSizeFunctions.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.LiteralParameters;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import io.airlift.slice.Slice;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.spi.type.Decimals.encodeUnscaledValue;
+import static java.lang.Character.isDigit;
+import static java.lang.String.format;
+
+public class DataSizeFunctions
+{
+    private DataSizeFunctions() {}
+
+    @Description("converts data size string to bytes")
+    @ScalarFunction("parse_data_size")
+    @LiteralParameters("x")
+    @SqlType("decimal(38,0)")
```

Comment:
Nit: `decimal(38)`?

Reply:
No supported =\

Reply:
Interesting error:
```
Query 20181106_232125_00002_5w3xa failed: parse_data_size(varchar(3)):decimal(38) not found
```
I suspect this is a bug, but that's a problem for another day.